apps/wallet-dashboard/components/migration/migration-object-details-card/MigrationObjectDetailsCard.tsx

```diff
@@ -0,0 +1,146 @@
+                />
+            );
+        case CommonMigrationObjectType.Nft:
+            return (
+                <MigrationObjectCard
+                    title={migrationObject.name}
+                    subtitle="Visual Assets"
+                    unlockConditionTimestamp={unlockConditionTimestamp}
+                    image={
+                        <ExternalImageWithFallback
+                            src={migrationObject.image_url}
+                            alt={migrationObject.name}
+                            fallback={<Assets />}
+                        />
+                    }
+                    isTimelocked={isTimelocked}
+                />
+            );
+        case CommonMigrationObjectType.NativeToken:
+            return (
+                <MigrationObjectCard
+                    isTimelocked={isTimelocked}
+                    title={`${balance} ${token}`}
+                    subtitle="Native Tokens"
+                    unlockConditionTimestamp={unlockConditionTimestamp}
+                    image={<DataStack />}
+                />
+            );
+        default:
+            return null;
+    }
+}
+
+interface ExternalImageWithFallbackProps {
+    src: string;
+    alt: string;
+    fallback: React.ReactNode;
+}
+function ExternalImageWithFallback({ src, alt, fallback }: ExternalImageWithFallbackProps) {
+    const [errored, setErrored] = useState(false);
+    function handleError() {
+        setErrored(true);
+    }
+    return !errored ? <ExternalImage src={src} alt={alt} onError={handleError} /> : fallback;
+}
+
+interface MigrationObjectCardProps {
+    title: string;
+    subtitle: string;
+    unlockConditionTimestamp: string;
+    isTimelocked: boolean;
+    image?: React.ReactNode;
+}
+
+function MigrationObjectCard({
+    title,
+    subtitle,
+    unlockConditionTimestamp,
+    isTimelocked,
+    image,
+}: MigrationObjectCardProps) {
+    const hasUnlockConditionTimestamp =
+        unlockConditionTimestamp !== MIGRATION_OBJECT_WITHOUT_UC_KEY;
+    return (
+        <Card>
+            <CardImage shape={ImageShape.SquareRounded}>{image}</CardImage>
+            <CardBody title={title} subtitle={subtitle} />
+            {hasUnlockConditionTimestamp && (
+                <UnlockConditionLabel
+                    groupKey={unlockConditionTimestamp}
+                    isTimelocked={isTimelocked}
+                />
+            )}
+        </Card>
+    );
+}
+
+interface UnlockConditionLabelProps {
+    groupKey: string;
+    isTimelocked: boolean;
+}
+function UnlockConditionLabel({ groupKey, isTimelocked: isTimelocked }: UnlockConditionLabelProps) {
+    const { data: currentEpochEndTimestampMs } = useGetCurrentEpochEndTimestamp();
+
+    const epochEndMs = currentEpochEndTimestampMs ?? 0;
+    const currentDateMs = Date.now();
+
+    const unlockConditionTimestampMs = Number(groupKey) * 1000;
+    const isInAFutureEpoch = unlockConditionTimestampMs > epochEndMs;
```

Comment:
This is temporary feaute, we could add a TODO comment to this [issue](https://github.com/iotaledger/iota/issues/4369) to modify it when possible